xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.cpp

```diff
@@ -394,6 +401,17 @@ void CActiveAESink::StateMachine(int signal, Protocol *port, Message *msg)
           m_volume = *(float*)msg->data;
           m_sink->SetVolume(m_volume);
           return;
+
+        case CSinkControlProtocol::NOISETYPECHANGED:
+        {
+          bool streamNoise = *(bool*)msg->data;
+          if (streamNoise != m_streamNoise)
```

Comment:
Yes. You are right here. It basically is a "triggered" signal. With a bool it will always change - so this is most likely not needed here, right? But think of e.g. streamsilence parameter, where user could pick up the setting, scroll through it and select the same value as before.